packages/openapi-ts/src/index.ts

```diff
@@ -228,6 +237,12 @@ const initConfigs = async (userConfig: UserConfig): Promise<Config[]> => {
     const services = getServices(userConfig);
     const types = getTypes(userConfig);
 
+    const transform = getTransforms(userConfig);
+    if (transform.dates) {
+      // If transforming dates then also force the type on
+      types.dates = true;
```

Comment:
Honestly we should only have 1 config, neither makes much sense without the other one

Reply:
Do you think a 3-state value with backward compat (ie. `date: boolean | "transform" | "types-only"`) for this option makes sense, or when `dates: true` is set we just do the transform automatically and accept that as a breaking change?

Reply:
I am fine with breaking changes as you may already know + this would be a simple migration for users. 3-state sounds great, maybe values could be a bit different, though each value will be explained in JSDoc comment too. I am thinking `types` instead of `types-only` and maybe `types+transform` instead of `transform` or `full` or something to that extent?

Reply:
Sounds perfect, I can change that